### DISCUSSION TITLE
Add a virtual bot to the server, which will be displayed as a player

### DISCUSSION THREAD

**Discussion by barashF** [SmartlyDressedGames/Unturned-3.x-Community#4022] [Q&A] [answered]
I am thinking about how to add a virtual account to the server, which will be displayed as a player. I thought about running Steam accounts in parallel on the server itself and logging in, but all this is very cumbersome, maybe there is an easier option, I will be grateful for the push in the right direction, but now I'm at a dead end

**Comment by GazziFX** [comment, marked as answer]
Look at player connecting

**Reply by barashF** [reply]
Have you had any experience with this? I just went through the Provider completely, tried to add a new user through accept, but there are some troubles

**Reply by GazziFX** [reply]
Yes I have working bots on server, there are many fields that you need to initialize manually

**Reply by barashF** [reply]
Would it be impudent of me to ask for a hint?) Now I just thought about implementing it through cloud servers, because I don’t see any other options

**Reply by GazziFX** [reply]
```csharp
var pending = new SteamPending(new your_own_transport_connection_class(), other properties...);

// inventory details and etc
pending.some_fields = ...;

Provider.pending.Add(pending); // add to queue
Provider.accept(pending);
```

**Comment by sunnamed434** [comment]
Look at the [Dummy](https://github.com/EvolutionPlugins/Dummy) plugin, you also need to fake the bot ping and patch one method against the batteye (to prevent bots kicking), you can find all of those things in this repo, also you would probably see a problem that bots won't be displayed in the server game menu, this is more about the Steam API questions, you will probably need to make bypass for it

**Reply by barashF** [reply]
Thanks for the tip, I'll take a look
